components/tests/ui/testcases/web/create_scenario.txt

```diff
@@ -0,0 +1,298 @@
+    Wait Until Page Contains Element		xpath=//tr[contains(@class,'data_heading_id')]//th[contains(text(),'${containerOption} ID:')]
+    ${Status}=								Run Keyword and Return Status 		 Wait Until Page Contains Element	${XPathNoData}	
+    Run Keyword If 							'${containerOption}'=='Dataset' and ${Status} == 'FAIL'  				Wait Until Page Contains Element  	xpath=//ul[@id='dataIcons' and contains(@class,'ui-selectable')]		    			
+    ${newId}    							Get Text       	css=tr.data_heading_id strong
+    [Return]								${newId}
+
+Create Project Using Icon and Check If It Exists
+	${projectId}=							Create Project 						${CtxProjectName}
+	Project Should Exist					${projectId}
+
+Create Project Using Right Click and Check If It Exists
+	[Arguments]								${RootId}							${ProjectName}
+	Right Click Create P/D/S 				${RootId}		Project				${ProjectName}
+	${projectId}=							Check Right And Center Panels For Active Container 						Project 			${ProjectName}
+	Project Should Exist					${projectId}
+
+Project Should Exist
+	[Arguments]								${projectId}	
+	${nodeId}=								Wait for Project Node  				${projectId}
+    Wait Until Page Contains Element		xpath=//li[@id='${treeRootId}']/ul/li[@id='${nodeId}']
+
+Create Dataset Using Icon and Check If Orphaned
+	[Arguments]								${isOrphaned}
+	${projectId}= 							Run Keyword If						'${isOrphaned}' == '${notOrphaned}'	Get Text    		css=tr.data_heading_id strong
+	${datasetId}=							Create Dataset 						${CtxDatasetName}
+	Run Keyword If 							'${isOrphaned}' == '${orphaned}'	Dataset Exists and Is Orphaned		${datasetId}
+	Run Keyword If 							'${isOrphaned}' == '${notOrphaned}'	Dataset Exists and Is Not Orphaned 	${datasetId}		${projectId}
+
+Create Dataset Using Right Click and Check If Orphaned
+	[Arguments]								${RootId}		${DatasetName}		${isOrphaned}
+	${projectId}= 							Run Keyword If						'${isOrphaned}' == '${notOrphaned}'	Get Text    		css=tr.data_heading_id strong
+	Right Click Create P/D/S 				${RootId}		Dataset				${DatasetName}
+	${datasetId}=							Check Right And Center Panels For Active Container 						Dataset 			${DatasetName}
+	Run Keyword If 							'${isOrphaned}' == '${orphaned}'	Dataset Exists and Is Orphaned		${datasetId}
+	Run Keyword If 							'${isOrphaned}' == '${notOrphaned}'	Dataset Exists and Is Not Orphaned 	${datasetId}		${projectId}	
+
+Dataset Exists and Is Orphaned
+	[Arguments]								${datasetId}	
+	${nodeId}=								Wait for Dataset Node  				${datasetId}
+    Wait Until Page Contains Element		xpath=//li[@id='${treeRootId}']/ul/li[@id='${nodeId}']
+
+Dataset Exists and Is Not Orphaned
+	[Arguments]								${datasetId}						${projectId}
+	${datasetnodeId}=						Wait for Dataset Node 				${datasetId}
+	${projectnodeId}=						Wait for Project Node 				${projectId}
+	Wait Until Page Contains Element		xpath=//li[@id='${treeRootId}']/ul/li[@id='${projectnodeId}']/ul/li[@id='${datasetnodeId}']
+
+Dataset Should Contain Image 
```

Comment:
This keyword should probably take DatasetId as an argument and simply click the Dataset and wait until the Image with the correct Id appears as a child in the tree (is a child of Dataset node). Then this keyword would be more generally useful (as you suggest.
If you return the DatasetId from your previous keyword ```Create Dataset Using Right Click and Check If Orphaned``` then you'll have that in hand and won't have to look it up.